demucs/api.py

```diff
@@ -0,0 +1,834 @@
+            will be stored on `wav.device`. If not specified, will use the command line option.
+        num_workers: Number of jobs. This can increase memory usage but will be much faster when \
+            multiple cores are available. If not specified, will use the command line option.
+        callback: A function will be called when the separation of a chunk starts or finished. \
+            The argument passed to the function will be a dict. For more information, please see \
+            the Callback section.
+        callback_arg: A dict containing private parameters to be passed to callback function. For \
+            more information, please see the Callback section.
+        progress: If true, show a progress bar.
+
+        Returns
+        -------
+        Separated stems.
+
+        Callback
+        --------
+        The function will be called with only one positional parameter whose type is `dict`. The
+        `callback_arg` will be combined with information of current separation progress. The
+        progress information will override the values in `callback_arg` if same key has been used.
+
+        Progress information contains several keys (These keys will always exist):
+        - `model_idx_in_bag`: The index of the submodel in `BagOfModels`. Starts from 0.
+        - `shift_idx`: The index of shifts. Starts from 0.
+        - `segment_offset`: The offset of current segment. If the number is 441000, it doesn't
+            mean that it is at the 441000 second of the audio, but the "frame" of the tensor.
+        - `state`: Could be `"start"` or `"end"`.
+        - `audio_length`: Length of the audio (in "frame" of the tensor).
+        - `models`: Count of submodels in the model.
+
+        Notes
+        -----
+        Use this function with cautiousness. This function does not provide data verifying.
+        """
+        ref = wav.mean(0)
+        wav -= ref.mean()
+        wav /= ref.std()
+        ret = self._separate_track(
+            wav[None],
+            model=model,
+            segment=segment,
+            shifts=shifts,
+            split=split,
+            overlap=overlap,
+            transition_power=transition_power,
+            device=device,
+            num_workers=num_workers,
+            pool=pool,
+            callback=callback,
+            callback_arg=_replace_dict(callback_arg, ("audio_length", wav.shape[1])),
+            progress=progress,
+        )
+        ret *= ref.std()
+        ret += ref.mean()
+        return ret
+
+    def separate_loaded_audio(
+        self,
+        segment: Optional[float] = 0.0,
+        shifts: Optional[int] = None,
+        split: Optional[bool] = None,
+        overlap: Optional[float] = None,
+        transition_power=1.0,
+        device=None,
+        num_workers=None,
+        callback: Optional[Callable[[dict], None]] = None,
+        callback_arg: Optional[dict] = None,
+        progress: bool = False,
```

Comment:
I wouldn't allow for overriding the separation params inside each function of the Separator. I think most of the time it doesn't make sense and it makes the code more complex and less compact and more bug prone (what happens if in one function you forgot to pass one of the param?). At best you could have a single `update_params()` methods that would update the params in  place and if people really need it they can. 

My own philosophy is that you should aim for covering most of the use case with the simplest possible API. the remaining 0.1% of the use case (e.g. people want to test different params without reloading the model) are honestly not worth it, as most of the time there would be a single set of params that is optimal for a use case.